engine/src/test/java/com/redhat/cloud/notifications/events/EndpointProcessorTest.java

```diff
@@ -318,4 +282,104 @@ void testEndpointRestrictionAccordingEventType(final boolean isEventTypeRestrict
         }
     }
 
+    /**
+     * Tests Event-Driven Ansible endpoint type being aliased to a WebHook processor.
+     */
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    void testBlacklistedEndpoints(final boolean useEndpointToEventTypeDirectLink) {
+        // Create an Endpoint which will be simulated to be fetched from the database.
+        final String orgId = "test-org-id";
+        final UUID endpointUuid = UUID.randomUUID();
+        final UUID endpointUuid2 = UUID.randomUUID();
+        Mockito.when(engineConfig.isUseDirectEndpointToEventTypeEnabled()).thenReturn(useEndpointToEventTypeDirectLink);
+
+        final Endpoint blacklistedEndpoint = new Endpoint();
+        blacklistedEndpoint.setId(endpointUuid);
+        blacklistedEndpoint.setOrgId(orgId);
+        blacklistedEndpoint.setType(EndpointType.ANSIBLE);
+
+        final Endpoint regularEndpoint = new Endpoint();
+        regularEndpoint.setId(endpointUuid2);
+        regularEndpoint.setOrgId(orgId);
+        regularEndpoint.setType(EndpointType.WEBHOOK);
+
+        final Endpoint systemEndpoint = new Endpoint();
+        systemEndpoint.setId(endpointUuid2);
+        systemEndpoint.setType(EndpointType.EMAIL_SUBSCRIPTION);
+
+        Action action = buildAction(orgId);
+
+        final EventType eventType = new EventType();
+        eventType.setId(UUID.randomUUID());
+
+        Event event = new Event();
+        event.setId(UUID.randomUUID());
+        event.setEventWrapper(new EventWrapperAction(action));
+        event.setEventType(eventType);
+        event.setOrgId(orgId);
+
+        Mockito.when(this.endpointRepository.getTargetEndpoints(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint, systemEndpoint));
+        Mockito.when(this.endpointRepository.getTargetEndpointsWithoutUsingBgs(Mockito.anyString(), Mockito.any(EventType.class))).thenReturn(List.of(regularEndpoint, blacklistedEndpoint, systemEndpoint));
+
+        Mockito.doNothing().when(this.webhookProcessor).process(Mockito.any(Event.class), Mockito.anyList());
```

Comment:
Is `doNothing` really needed? `webhookProcessor` is mocked so it should already do nothing by default.